mne/io/eeglab/eeglab.py

```diff
@@ -226,6 +233,12 @@ class RawEEGLAB(_BaseRaw):
         Names or indices of channels that should be designated
         EOG channels. If 'auto', the channel names containing
         ``EOG`` or ``EYE`` are used. Defaults to empty tuple.
+    event_id : dict
+        By default, events are read from the input file by dropping every
+        non-integer part of events containing integers, and completely
+        dropping any events without integer parts. If non-integer events
+        should be read, this should be a dict mapping from their names to
+        integers, e.g. dict(fmri_scan_onset=199, recording_start=255).
```

Comment:
Examples:

The EEGLAB files I am looking at typically have events with triggers like 'S196', 'S  1', 'S  2', , ..., 'S100', 'recording pause' (or with a D instead of the S for other sources). Currently, this would be parsed as 196, 1, 2, ..., 100, and the string-only one would be dropped with a warning. To keep it, you'd have to use the event_id dict - this is what we're using in the brainvision reader.

Basically, the way it currently works means if you read a BV or EGI file using MNE directly, and if you read it into EEGLAB, export it to EEGLAB format, and read that with MNE, you get the same results.
(For both cases, if you want to keep e.g. 'fmri_scan_onset', you have to use the event_id dict.)

Based on these usage scenarios, option 1 seems very undesirable because it requires a lot of work of the user. For example, imagine half of your files have a trigger called 'recording pause', and half of them don't. In that case, for the first half, 'S  1' would be mapped to 1, but for the other half, 'S  1' would be mapped to 2. Moreover, you'd have to run the reader to figure out which is which. That's a nightmare scenario IMO.

Option 2 I don't like because we don't need 2 stages/accessing the original file twice for the other readers, which we don't have anywhere else do we? And it has more undesirable properties. For example, usually, you'll define a very similar event_id dict later on anyways - for example, you'll map 'S  1'/1 to "button/left", 'S  2' to "button/right" and so on. So you have two large event_id dicts that look extremely redundant.

Option 3 is just tons of manual work that we don't, I think, need. If you have 150 types of triggers of the form described above, that means tons of manual work when a simple parser would do.

I like @jasmainak 's idea of giving a callable to transform event codes, especially if we can set it up so that it has sensible, realistic defaults (like my preferred "Drop the non-integer parts and keep the ints" :)/string-associated methods).